client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/IndicesClientDocumentationIT.java

```diff
@@ -771,6 +773,76 @@ public void onFailure(Exception e) {
             }
             // end::flush-notfound
         }
+    }
+
+    public void testGetSettings() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+
+        {
+            Settings settings = Settings.builder().put("number_of_shards", 3).build();
+            CreateIndexResponse createIndexResponse = client.indices().create(new CreateIndexRequest("index", settings));
+            assertTrue(createIndexResponse.isAcknowledged());
+        }
+
+        // tag::get-settings-request
+        GetSettingsRequest request = new GetSettingsRequest().indices("index");
+        // tag::get-settings-request
+
+        // tag::get-settings-request-names
+        request.names("index.number_of_shards"); // <1>
+        // end::get-settings-request-names
+
+        // tag::get-settings-request-masterTimeout
+        request.masterNodeTimeout(TimeValue.timeValueMinutes(1)); // <1>
+        request.masterNodeTimeout("1m"); // <2>
+        // end::get-settings-request-masterTimeout
+
+        // tag::get-settings-request-indicesOptions
+        request.indicesOptions(IndicesOptions.lenientExpandOpen()); // <1>
+        // end::get-settings-request-indicesOptions
+
+
+        // tag::get-settings-execute
+        GetSettingsResponse getSettingsResponse = client.indices().getSettings(request);
+        // end::get-settings-execute
+
+        // tag::get-settings-response
+        String numberOfShardsString = getSettingsResponse.getSetting("index", "index.number_of_shards"); // <1>
+        Settings indexSettings = getSettingsResponse.getIndexToSettings().get("index"); // <2>
+        int numberOfShards = indexSettings.getAsInt("index.number_of_shards", null); // <3>
```

Comment:
this is not a great example I'm afraid, as you set a default value to `null`, but given that the variable is `int` a null pointer will be thrown